halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/OrcaProfileFactory.java

```diff
@@ -80,14 +81,18 @@ protected void setProfile(
     // For backward compatibility
     profile.appendContents("pipelineTemplate.enabled: " + pipelineTemplates);
 
+    // Plugins
     final List<Plugin> plugins = deploymentConfiguration.getPlugins().getPlugins();
     Map<String, Object> fullyRenderedYaml = new LinkedHashMap<>();
     Map<String, Object> pluginMetadata =
         plugins.stream()
             .filter(p -> p.getEnabled())
             .filter(p -> !p.getManifestLocation().isEmpty())
-            .map(p -> p.generateManifest())
-            .collect(Collectors.toMap(m -> m.getName(), m -> m.getOptions()));
+            .map(p -> new AbstractMap.SimpleEntry<>(p, p.generateManifest()))
```

Comment:
This construct is somewhat confusing to me---we're creating a temporary `AbstractMap.SimpleEntry` in the stream, but then are collecting to a map that doesn't actually use these entries.  (The indirection also makes it hard to see what we're calling `getOptions()` on below as it's not immediately clear what the value and key are.)

Given that the `Plugin` class has access to all the data it needs to compute the merged options, it might be more clear to have it expose a function to return these merged options (maybe even just doing this in `generateManifest()`) instead of requiring callers to send back data through a static method.

Reply:
ah thats a good call, moving 👍